nav2_bringup/launch/tb3_gz_robot_launch.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (C) 2023 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ament_index_python.packages import get_package_share_directory
+
+from launch import LaunchDescription
+from launch.actions import AppendEnvironmentVariable
+from launch.actions import DeclareLaunchArgument
+from launch.conditions import IfCondition
+from launch.substitutions import LaunchConfiguration
+
+from launch_ros.actions import Node
+
+
+def generate_launch_description():
+    bringup_dir = get_package_share_directory('nav2_bringup')
+
+    namespace = LaunchConfiguration('namespace')
+    use_simulator = LaunchConfiguration('use_simulator')
+    robot_name = LaunchConfiguration('robot_name')
+    robot_sdf = LaunchConfiguration('robot_sdf')
+    pose = {'x': LaunchConfiguration('x_pose', default='-2.00'),
+            'y': LaunchConfiguration('y_pose', default='-0.50'),
+            'z': LaunchConfiguration('z_pose', default='0.01'),
+            'R': LaunchConfiguration('roll', default='0.00'),
+            'P': LaunchConfiguration('pitch', default='0.00'),
+            'Y': LaunchConfiguration('yaw', default='0.00')}
+
+    # Declare the launch arguments
+    declare_namespace_cmd = DeclareLaunchArgument(
+        'namespace',
+        default_value='',
+        description='Top-level namespace')
+
+    declare_use_simulator_cmd = DeclareLaunchArgument(
+        'use_simulator',
+        default_value='True',
+        description='Whether to start the simulator')
+
+    declare_robot_name_cmd = DeclareLaunchArgument(
+        'robot_name',
+        default_value='turtlebot3_waffle',
+        description='name of the robot')
+
+    declare_robot_sdf_cmd = DeclareLaunchArgument(
+        'robot_sdf',
+        default_value=os.path.join(bringup_dir, 'worlds', 'gz_waffle.sdf'),
+        description='Full path to robot sdf file to spawn the robot in gazebo')
+
+    pkg_nav2_bringup = get_package_share_directory('nav2_bringup')
+    bridge = Node(
```

Comment:
Shouldn't this bridge also be namespaced?

Reply:
Not sure what you mean. The topics in the Gazebo classic model are not namespaced (e.g., Laser scan is published on `/scan`).

Reply:
Not when we launch using the `namespace`, then multiple robots are possible to be simulated and it publishes to `namespace/scan` -- see the multi-robot launch files in this same directory (unique [which are N unique robots] / clone [which are N of the same robot]). So, I'd think the bridge would need to have the namespace applied to have the topic broadcast on the set namespace, no?

Reply:
Ah, I wasn't aware of the multi-robot launch files. I think the bridge and the sdf files need to be reworked to support multiple robots. Just adding the `namespace` here won't be enough since the `gz-transport` topics also need to be "namespaced". The `turtlebot3_waffle_bridge.yaml` file will also have to become some sort of template so we can inject the robot name. 

Also, those multirobot launch files refer to the Gazebo classic based launch files. Should we copy them for the modern Gazebo versions?

I probably won't be able to get to this till later next week.